gnovm/stdlibs/testing/match.gno

```diff
@@ -166,19 +166,11 @@ func isSpace(r rune) bool {
 	return false
 }
 
-var (
-	matchPat string
-	matchRe  *regexp.Regexp
-)
-
 // based on testing/internal/testdeps.TestDeps.MatchString.
 func matchString(pat, str string) (result bool, err error) {
-	if matchRe == nil || matchPat != pat {
-		matchPat = pat
-		matchRe, err = regexp.Compile(matchPat)
-		if err != nil {
-			return
-		}
+	matchRe, err := regexp.Compile(pat)
+	if err != nil {
+		return
```

Comment:
As mentioned in PR body, with a test that is now considered as a realm, we can no longer have global variables in dependency packages.